src/core/symbol/SymbolTransaction.ts

```diff
@@ -0,0 +1,357 @@
+            transactions: embeddedBuilders,
+            cosignatures: [],
+        });
+        const builder = SymbolTransactionUtils.createFromBodyBuilder({
+            fee: new AmountDto(fee),
+            deadline: new TimestampDto(BigInt(deadline.adjustedValue)),
+            type: type,
+            network: this.network.identifier,
+            bodyBuilder: bodyBuilder,
+        });
+        return this.createFromBuilder(builder);
+    }
+
+    /**
+     * It converts an Body Builder (e.g: TransferTransactionBodyBuilder) to an Embedded Builder (e.g: EmbeddedTransferTransactionBuilder)to be used in an aggregate transaction
+     * @param bodyBuilder - the body builder
+     * @param signerPublicKey - the signer public key.
+     */
+    public toEmbedded(bodyBuilder: Serializer, signerPublicKey: Key): EmbeddedTransactionBuilder {
+        return SymbolTransactionUtils.createEmbeddedFromBodyBuilder({
+            network: this.network.identifier,
+            signerPublicKey: new KeyDto(signerPublicKey.key),
+            bodyBuilder: bodyBuilder,
+        });
+    }
+
+    /**
+     * Helper method that serializes an unresolved address to be used in Catbuffer builders.
+     * @param unresolvedAddressId - the unresolved address to serialize.
+     */
+    public toUnresolvedAddress(unresolvedAddressId: SymbolUnresolvedAddress): Uint8Array {
+        if (typeof unresolvedAddressId == 'bigint') {
+            return SymbolIdGenerator.encodeUnresolvedAddress(this.network.identifier, unresolvedAddressId);
+        } else {
+            return unresolvedAddressId.getAddressBytes();
+        }
+    }
+
+    /**
+     * Generic method to create transaction from a top level (simple or aggregate) builder.
+     * @param builder - the builder
+     */
+    public createFromBuilder<T extends TransactionBuilder>(builder: T): SymbolTransaction<T> {
+        return SymbolTransaction.createFromBuilder(this.network, builder);
+    }
+
+    /**
+     * Generic method to create transaction from a top level (simple or aggregate) payload.
+     *
+     * @param payload - the serialized payload.
+     */
+    public createFromPayload(payload: Uint8Array): SymbolTransaction {
+        return SymbolTransaction.createFromPayload(this.network, payload);
+    }
+
+    /**
+     * Hides how transactions hash is generated from the provided embedded transactions. Dev doesn't need to calculate at front.
+     * @param embeddedBuilders - the builders of the embedded transactions.
+     */
+    private calculateAggregateTransactionsHashFromPayloads(embeddedBuilders: EmbeddedTransactionBuilder[]): Uint8Array {
+        return SymbolTransactionUtils.calculateAggregateTransactionsHashFromPayloads(embeddedBuilders.map((b) => b.serialize()));
+    }
+}
+
+/**
+ * The mutable generic symbol transaction that wraps the signing over catbuffer builders.
+ */
+export class SymbolTransaction<T extends TransactionBuilder = TransactionBuilder> {
```

Comment:
This file is getting too big. It would be better to split SymbolTransaction and TxFactory into individual classes, following TS one file per class pattern.

Reply:
I have split the file